src/tests/ftest/telemetry/telemetry_pool_space_metrics.py

```diff
@@ -0,0 +1,148 @@
+                "engine_pool_vos_space_nvme_used": (self.data_size, self.data_size)
+            }
+        }
+
+    def get_expected_values_range(self, namespace):
+        """Return the expected metrics value output.
+
+        This function returns a hash map of pairs defining min and max values of each tested
+        telemetry metrics.  The hash map of pairs returned depends on the pool created with the
+        given namespace and the size of the data written in it.
+
+        Args:
+            namespace (string): Namespace of the last created pool.
+
+        Returns:
+            expected_values (dict): Dictionary of the expected metrics value output.
+        """
+
+        self.assertIn(
+            namespace, self.pool_space_metrics_minmax,
+            "Invalid pool namespace: {}".format(namespace))
+
+        return self.pool_space_metrics_minmax[namespace]
+
+    def get_metrics(self, names):
+        """Obtain the specified metrics information.
+
+        Args:
+            name (list): List of metric names to query.
+
+        Returns:
+            metrics (dict): a dictionary of metric keys linked to their aggregated values.
+        """
+        metrics = {}
+        for name in names:
+            metrics[name] = 0
+
+        for data in self.telemetry.get_metrics(",".join(names)).values():
+            for name, value in data.items():
+                for metric in value["metrics"]:
+                    metrics[name] += metric["value"]
+
+        return metrics
+
+    def test_telemetry_pool_space_metrics(self):
+        """JIRA ID: DAOS-10192
+
+        Create a file of 16MiB thanks to ior to verify the DAOS engine IO telemetry vos space
+        metrics.
+
+        Steps:
+            Create a pool
+            Create a container
+            Generate deterministic workload.  Using ior to write 128MiB of data.
+            Use telemetry command to get value of vos space metrics "engine_pool_vos_space_scm_used"
+            and "engine_pool_vos_space_nvme_used" for all targets.
+            Verify the sum of all parameter metrics matches the workload.
+
+        :avocado: tags=all,daily_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=telemetry
+        :avocado: tags=telemetry_pool_space_metrics,test_telemetry_pool_space_metrics
+        """
+
+        for namespace in ["/run/pool_scm/*", "/run/pool_scm_nvme/*"]:
+            # create pool and container
+            self.add_pool(namespace=namespace, create=True, connect=False)
+            self.pool.disable_aggregation()
+            self.add_container(pool=self.pool)
+
+            # Run ior command.
+            try:
+                self.update_ior_cmd_with_pool(create_cont=False)
+                self.run_ior_with_pool(
+                    timeout=200, create_pool=False, create_cont=False)
+            except TestFail as error:
+                self.log.info("ior command failed: %s", str(error))
+                raise
```

Comment:
`run_ior_with_pool` should log the failure, so this shouldn't be needed. Were you seeing different behavior?

Reply:
I have to admit that I have copy/pasted this part of the code from https://github.com/daos-stack/daos/blob/f45b589bd0467e2fa6d8a11ed4068c32ecbcfa36/src/tests/ftest/telemetry/telemetry_pool_metrics.py#L160

- [ ] Remove useless exception catch

Reply:
Fixed with commit 77f5e57